src/Model/Behavior/FootprintBehavior.php

```diff
@@ -68,22 +68,51 @@ public function initialize(array $config)
     }
 
     /**
-     * Injects configured fields into finder conditions.
-     *
-     * @param \Cake\Event\Event $event Event.
-     * @param \Cake\ORM\Query $query Query.
-     * @param \ArrayObject $options Options.
+     * {@inheritDoc}
+     */
+    public function implementedEvents()
+    {
+        /* map all configured events to a single handler */
+        return array_map(function () { return 'dispatch'; }, this->_config['events']);
```

Comment:
@ADmad how to reformat this so the code checker doesn't complain and it does also not look ridiculous?